src/core_read.cpp

```diff
@@ -155,3 +156,86 @@ vector<unsigned char> ParseHexUV(const UniValue& v, const string& strName)
         throw runtime_error(strName+" must be hexadecimal string (not '"+strHex+"')");
     return ParseHex(strHex);
 }
+
+static bool ParseKeyTreeNode(const std::string &s, size_t &pos, KeyTreeNode& tree);
+static bool ParseKeyTreeCall(const std::string &s, size_t &pos, unsigned long* num, std::vector<KeyTreeNode>& children)
+{
+    if (s.size() == pos) return false;
+    if (s[pos] != '(') return false;
+    pos++;
+    int count = 0;
+    if (num) {
+        const char *ptr = &s[pos];
+        char *eptr = NULL;
+        *num = strtoul(ptr, &eptr, 10);
+        if (eptr == ptr) return false;
+        pos += eptr - ptr;
+        count++;
+    }
+    while (true) {
+        if (count) {
+            if (pos == s.size()) return false;
+            if (s[pos] == /*(*/')') {
+                pos++;
+                return true;
+            }
+            if (s[pos] != ',') return false;
+            pos++;
+        }
+        children.push_back(KeyTreeNode());
+        if (!ParseKeyTreeNode(s, pos, children.back())) return false;
+        count++;
+    }
+}
+
+static bool ParseKeyTreeNode(const std::string &s, size_t &pos, KeyTreeNode& tree)
+{
+    while (pos < s.size() && isspace(s[pos])) pos++;
```

Comment:
I'd like to have this line at the top of `ParseKeyTreeCall` and at the top of the `while` loop in that function. That allows to have whitespace before `(` and `,`.